### ISSUE TITLE
(touts): If you modify this implementation please run 

### ISSUE THREAD

**Issue by piascikj** [imdone/tensorflow#2321]
the BM_ImageNetSoftmaxFwd benchmark in nn_ops_test.cc.

softmax = exp(logits - max(logits along classes));

----
*opened via [imdone.io](https://imdone.io) from a code comment on [6ff94239ef72159d50edcba664c4081bb70b4fde](https://github.com/imdone/tensorflow/commit/6ff94239ef72159d50edcba664c4081bb70b4fde) by A. Unique TensorFlower nobody@tensorflow.org*

----

https://github.com/imdone/tensorflow/blob/8cb6e535f6ea14b380aa86d425169adad682cb8c/tensorflow/core/kernels/softmax_op_functor.h#L83-L91
